### PR TITLE
Add datatable events handlers

### PR DESCRIPTION
**Pull request by abberdeen**
Hi @JustinWilkinson,

I would like to complement the component you created
DataTables, and contribute to the development of your component.

I added a basic ability to handle events in datatables.js, with the help of this add-on there will be the ability to listen for the event of the "DataTables" group (https://datatables.net/reference/event/).

Also updated the documentation in the demo application of this component.

Please review and if everything is ok, please confirm my pull request. I will be glad to receive your feedback and make the necessary changes for this pull request.

Kind Regards,
abberdeen

### CODE REVIEW

File: DataTables.Blazor.Demo/Pages/Index.razor [hunk 1]
```diff
@@ -556,12 +671,20 @@ private record Weather
         }
     };
 
+    private string EventsLogs { get; set; }
+
+    void AddEventLog(string log)
+    {
+        EventsLogs +=  $"- {log}\n";
+        Task.Run(() => JS.InvokeVoidAsync("demo.scrollToBottom", "eventsLogs"));
```
Comment (JustinWilkinson) [inline DataTables.Blazor.Demo/Pages/Index.razor]: Any reason not to make this method async and return the task?

File: DataTables.Blazor/DataTable.razor [hunk 2]
```diff
@@ -126,4 +278,81 @@
     public ValueTask DisposeAsync() => Datatables.DestroyAsync(_tableReference);
 
     private bool IsDomSourced() => string.IsNullOrWhiteSpace(SourceUrl) && Data == null;
+
+    private async Task AddEventListenersBeforeInitAsync()
+    {
+        if (OnInit.HasDelegate)
```
Comment (JustinWilkinson) [inline DataTables.Blazor/DataTable.razor:285]: This is something of a personal preference, but braces can make life easier when adding new lines and avoid silly errors like [this](https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/)

File: DataTables.Blazor/Events/CellClickEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text; 
+
+namespace DataTables.Blazor.Events
+{
+    public class CellClickEventArgs : EventArgs
+    {
+        public int ColumnId { get; set; }
+        public string? RowIndex { get; set; }
```
Comment (JustinWilkinson) [inline DataTables.Blazor/Events/CellClickEventArgs.cs]: Would you mind fixing these warnings?

File: DataTables.Blazor/Interop/DomEventListener.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+        /// <summary>
+        /// Remove event listener by name
+        /// </summary>
+        /// <param name="dom"></param>
+        /// <param name="eventName"></param>
+        void Remove(object dom, string eventName);
+    }
+
+    public class DomEventListener : IDomEventListener
```
Comment (JustinWilkinson) [inline DataTables.Blazor/Interop/DomEventListener.cs]: Could be worth making this and the `Invoker` `internal`

File: DataTables.Blazor/Interop/DomEventListener.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        /// <param name="dom"></param>
+        /// <param name="eventName"></param>
+        /// <param name="callback"></param>
+        /// <returns></returns>
+        Task AddAsync<T>(ElementReference dom, string eventName, Action<T> callback);
+
+        /// <summary>
+        /// Clear all listeners and dispose.
+        /// </summary>
+        void Dispose();
```
Comment (JustinWilkinson) [inline DataTables.Blazor/Interop/DomEventListener.cs:35]: Probably worth this inheriting from `IDisposable`

File: DataTables.Blazor/DataTable.razor (hunk 2 above)
Comment (JustinWilkinson) [reply DataTables.Blazor/DataTable.razor:285]: It might also be worth covering these in a test?

**Review by JustinWilkinson — COMMENTED** [review]
Hey - this looks great, there's so much new functionality here! Just have a few comments.

File: DataTables.Blazor.Demo/Pages/Index.razor (hunk 1 above)
Comment (abberdeen) [reply DataTables.Blazor.Demo/Pages/Index.razor]: I usually use void for event handlers.

File: DataTables.Blazor/DataTable.razor (hunk 2 above)
Comment (abberdeen) [reply DataTables.Blazor/DataTable.razor:285]: Done

File: DataTables.Blazor/Events/CellClickEventArgs.cs (hunk 3 above)
Comment (abberdeen) [reply DataTables.Blazor/Events/CellClickEventArgs.cs]: Done

File: DataTables.Blazor/Interop/DomEventListener.cs (hunk 4 above)
Comment (abberdeen) [reply DataTables.Blazor/Interop/DomEventListener.cs]: Done

File: DataTables.Blazor/Interop/DomEventListener.cs (hunk 5 above)
Comment (abberdeen) [reply DataTables.Blazor/Interop/DomEventListener.cs:35]: Done

**Review by JustinWilkinson — APPROVED** [review]

**Pull request merged**